### PR TITLE
fix error while uninstalling Linuxbrew on CentOS 7

### PR DESCRIPTION
**Pull request by vtta**

### CODE REVIEW

File: uninstall [hunk 1]
```diff
@@ -264,7 +264,9 @@ def sudo?
   return @have_sudo unless @have_sudo.nil?
 
   Kernel.system "/usr/bin/sudo", "-v"
-  @have_sudo = $CHILD_STATUS&.success?
+  @have_sudo = $CHILD_STATUS.success?
```
Comment (sjackman) [inline uninstall:267]: `&.` is needed because `$CHILD_STATUS` may be `nil`.

File: uninstall [hunk 2]
```diff
@@ -264,7 +264,9 @@ def sudo?
...
+rescue Interrupt
```
Comment (sjackman) [inline uninstall:268]: Is this to rescue a keyboard Ctrl-C interrupt?

**Comment by sjackman** [comment]
Please fill out the GitHub description and describe the purpose of this PR. Include the exact error message that you were seeing without this PR, and the steps to reproduce the error.

**Comment by vtta** [comment]
## purpose
fix the error when uninstalling Linuxbrew on a CentOS7 with system ruby:
```
➜  ~ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/uninstall)"

-e:267: syntax error, unexpected '.'
  @have_sudo = $CHILD_STATUS&.success?
                              ^
```
Or manually running uninstall script:
```
➜  ~ ./uninstall -f
./uninstall:267: syntax error, unexpected '.'
  @have_sudo = $CHILD_STATUS&.success?

➜  ~ sudo ./uninstall -f
./uninstall:267: syntax error, unexpected '.'
  @have_sudo = $CHILD_STATUS&.success?
```

## reproduce
On a CentOS7 machine, with system ruby installed, running `/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/uninstall)"` directly or running uninstall script after manually downloading.

My environment: 
```
➜  ~ cat /etc/redhat-release
CentOS Linux release 7.6.1810 (Core)

➜  ~ /usr/bin/ruby --version
ruby 2.0.0p648 (2015-12-16) [x86_64-linux]

➜  ~ uname -a
Linux XXX 3.10.0-957.21.3.el7.x86_64 #1 SMP Tue Jun 18 16:35:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

## cause
Default ruby version on a CentOS7 machine is too low for the operator `&.`

## fix
Use `.` instead of `&.` according the similar code in install script
```
def sudo?
  return @have_sudo unless @have_sudo.nil?

  Kernel.system "/usr/bin/sudo", "-l", "mkdir"
  @have_sudo = $CHILD_STATUS.success?
rescue Interrupt
  exit
end
```
I am a little confused why it’s OK not to consider nil in the install script.

**Comment by sjackman** [comment]
`&.` is needed because `$CHILD_STATUS` may be `nil`, so I'm closing this PR. Thanks all the same.
If your host's version of Ruby is too old to run `uninstall`, you can uninstall Homebrew on Linux using
```sh
rm -r /home/linuxbrew/.linuxbrew ~/.linuxbrew ~/.cache/Homebrew
```

**Pull request closed without merging**